### PR TITLE
Added pruned, auto-gen code

### PR DESCRIPTION
**Pull request by michalpelka**
Autogen code for added. The actual functionality will be in follow-up PR.

### CODE REVIEW

File: Gems/LevelModificationTools/Code/Source/LevelModificationToolsModuleInterface.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Memory/Memory_fwd.h>
+#include <AzCore/Module/Module.h>
+#include <AzCore/RTTI/RTTIMacros.h>
+#include <AzCore/RTTI/TypeInfoSimple.h>
+
+namespace LevelModificationTools
```
Comment (pijaro) [inline Gems/LevelModificationTools/Code/Source/LevelModificationToolsModuleInterface.h:14]: A bit late I know, but wouldn't `EntityModificationTools` be a better name for that?

File: Gems/LevelModificationTools/gem.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+    "gem_name": "LevelModificationTools",
+    "version": "1.0.0",
+    "display_name": "LevelModificationTools",
+    "license": "Apache-2.0",
+    "license_url": "https://opensource.org/licenses/Apache-2.0",
+    "origin": "Robotec AI",
+    "origin_url": "https://github.com/RobotecAI/robotec-o3de-tools",
+    "type": "Code",
+    "summary": "Handy interface to replace / modify level in runtime.",
```
Comment (pijaro) [inline Gems/LevelModificationTools/gem.json:10]:
```suggestion
    "summary": "Handy interface to replace / modify entities in runtime.",
```

? :)

**Review by pijaro — COMMENTED** [review]
Isn't the `LevelModificationTools` directory duplicated?
1. First one is in `/Gems/LevelModificationTools`
2. Second one in `/LevelModificationTools`

**Comment by michalpelka** [comment]
Adjusted and removed duplicatation.

**Pull request merged**